### DISCUSSION TITLE
Can't disable readability-magic-numbers when all checks are enabled using *

### DISCUSSION THREAD

**Discussion by PerMalmberg** [clangd/clangd#2073] [Q&A] [answered]
Given this `.clang-tidy`, I'm expecting `readability-magic-number` to be disabled. That does not happen however, see screenshot.

```yaml
---
Checks: "*,
        -abseil-*,
        -altera-*,
        -android-*,
        -fuchsia-*,
        -google-*,
        -llvm*,
        -modernize-use-trailing-return-type,
        -zircon-*,
        -readability-else-after-return,
        -readability-static-accessed-through-instance,
        -readability-avoid-const-params-in-decls,
        -readability-magic-numbers,
        -cppcoreguidelines-non-private-member-variables-in-classes,
        -misc-non-private-member-variables-in-classes,
"
WarningsAsErrors: ''
HeaderFilterRegex: ''
FormatStyle:     none

CheckOptions:
  - key: readability-identifier-length.IgnoredVariableNames
    value: 'x|y|z'
  - key: readability-identifier-length.IgnoredParameterNames
    value: 'x|y|z'
```
![image](https://github.com/clangd/clangd/assets/2729388/f0b28c2d-9fcf-47e0-9f53-036ca51d79ed)

If I remove the leading `*` from `Checks` I can enable/disable the check as needed. The other checks are correctly disabled with `*` in place, so why not `readability-magic-numbers`?

**Comment by HighCommander4** [comment, marked as answer]
According to https://clang.llvm.org/extra/clang-tidy/checks/list.html, `readability-magic-numbers` has an alias called `cppcoreguidelines-avoid-magic-numbers`. My guess is you have to remove both names.

**Reply by PerMalmberg** [reply]
Huh, you're correct. I did see that alias but didn't think I had to disable *both*. Lesson learned, thank you!
